ci-operator/step-registry/ipi/conf/ibmcloud/provisionednetwork/ipi-conf-ibmcloud-provisionednetwork-commands.sh

```diff
@@ -6,9 +6,55 @@ set -o pipefail
 
 
 CONFIG="${SHARED_DIR}/install-config.yaml"
+vpcSubnetsFile="${SHARED_DIR}/customer_vpc_subnets.yaml"
 
-cat "${SHARED_DIR}/customer_vpc_subnets.yaml"
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output   
+    region="${LEASED_RESOURCE}"
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login..." 
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+    "${IBMCLOUD_CLI}" version
+    "${IBMCLOUD_CLI}" plugin list
+}
 
-yq-go m -x -i "${CONFIG}" "${SHARED_DIR}/customer_vpc_subnets.yaml"
+function  update_subnets() {
+  local subnet_type=$1
+  local zone_pattern=$(IFS=\|; echo "${zones[*]}")
 
+  # Update the YAML file using yq
+  yq-v4 e ".platform.ibmcloud.${subnet_type}Subnets |= map(select(test(\"(${zone_pattern})\")))" -i "$vpcSubnetsFile"
+}
+
+if [ ! -f "${vpcSubnetsFile}" ]; then
+    echo "${vpcSubnetsFile} is not found, exiting..."
+    exit 1
+fi
+
+cat "${vpcSubnetsFile}"
+
+ibmcloud_login
+
+readarray -t zones  < <(${IBMCLOUD_CLI} is zones -q | grep ${region} | awk '{print $1}')
```

Comment:
This make this step become much more complex than before, why not directly only saved the correct subnets when ZONES_COUNT is specified. 

And again, I wonder what is the issue for the original script before the introduce of  `ZONES_COUNT` ?